typing/typecore.ml

```diff
@@ -1904,7 +1904,7 @@ and type_pat_aux
       let p = type_pat tps category sp expected_ty' in
       let extra = (Tpat_constraint cty, loc, sp.ppat_attributes) in
       begin match category, (p : k general_pattern) with
-      | Value, {pat_desc = Tpat_var (id,s); _} ->
+      | Value, {pat_desc = Tpat_var (id,s); _} when false ->
```

Comment:
Did you really mean to include this?

Reply:
I didn't. Good catch!

(This is the code to disable the transformation in the type-checker, as discussed in #12522. I don't understand whether the transformation is necessary today and I don't mean to suggest it for inclusion for now. The present PR is a mitigation to ensure that the transformation does not break things.)